### PR TITLE
batch-system: Reduce the memory usage of peers' message channel 

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/16229

What's Changed:

```commit-message
Reduce the memory usage of peers' message channel 
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Test with 100k regions, the total memory usage reduce 40% compared to master
- patched: 
  - img_v3_02da_f6018662-1da3-4be4-9ddb-14ce1e963adg
- master:
  - img_v3_02da_4785f9d6-1845-4699-b2eb-78163f9e39cg

Gain 2% performance improvement on oltp_write_only 
a5caebbe-8ab2-4197-929d-0a4d64503d24

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Reduce considerable memory usage when there are lots of regions on one TiKV instance
```

### CODE REVIEW

**Comment by LykxSassinator** [comment]
rest LGTM, pls check and fix the remained CI errors.

**Review by hbisheng — APPROVED** [review]

File: components/raftstore/src/store/msg.rs [hunk 1]
```diff
@@ -795,28 +795,25 @@ impl fmt::Debug for InspectedRaftMessage {
 }
 
 /// Message that can be sent to a peer.
-#[allow(clippy::large_enum_variant)]
 #[derive(EnumCount, EnumVariantNames)]
 pub enum PeerMsg<EK: KvEngine> {
```
Comment (overvenus) [inline components/raftstore/src/store/msg.rs:799]: What's the current size of `PeerMsg`

**Review by glorv — APPROVED** [review]

**Review by LykxSassinator — APPROVED** [review]

**Pull request merged**